src/main/java/com/direwolf20/buildinggadgets/common/util/blocks/RegionSnapshot.java

```diff
@@ -119,11 +134,27 @@ private static NBTTagCompound serialize(NBTTagCompound tag, RegionSnapshot snaps
         return tag;
     }
 
-    private static RegionSnapshot deserialize(NBTTagCompound tag) {
+    public static RegionSnapshot deserialize(NBTTagCompound tag) {
         ResourceLocation dimension = new ResourceLocation(tag.getString(DIMENSION));
         World world = ServerLifecycleHooks.getCurrentServer().getWorld(Objects.requireNonNull(DimensionType.byName(dimension)));
 
-        Region region = Region.deserializeFrom(tag);
+        IPositionPlacementSequence positions;
+        if (tag.hasKey(POSITIONS_CLASS)) {
+            try {
```

Comment:
I don't think this is the best way to do this... Any ideas?

- `java.io.Serialzable`: This is used in the other branch, where all fields are serializable. However due to `BlockPos` not implementing `Serializable`, this would not work on `SetBackedPlacementSequence`

Reply:
If I remeber correctly you can define custom serilizers. You need to mark the BlockPos field transient and then implement read and write Object. After calling the defaultReadObject and defaultWriteObject to serialise all non transient fields, I presume there are Methods you can use to serialise the BlockPos.

Reply:
Or you could wrap BlockPos in something that implements Serialisable in the described way.